### PR TITLE
docs: adding explanation for color in `set_facecolor`

### PR DESCRIPTION
**Pull request by Borda**
## PR summary

Adding explanation what the color means in `set_facecolor`

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [x] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -1499,7 +1499,7 @@ def set_facecolor(self, color):
 
         Parameters
         ----------
-        color : color
+        color : color defined by string name, char abbreviation or hex, see :doc:`gallery/color/named_colors`
```
Comment (jklymak) [inline lib/matplotlib/axes/_base.py:1502]: So we have _many_ of these, where `color : color`.  

I think there is an argument for adding links: 

```
color : color; see :ref:`colors_def`
```
would be a better link (resolves to https://matplotlib.org/stable/users/explain/colors/colors.html).  

I know this was discussed before, but I forget the resolution.  @timhoffm is the API documentation lead.

Comment (timhoffm) [reply lib/matplotlib/axes/_base.py:1502]: See https://github.com/matplotlib/matplotlib/issues/24859#issuecomment-1368579653

Comment (jklymak) [reply lib/matplotlib/axes/_base.py:1502]: Right so I think the suggestion above is fine. 

Bonus points if this PR searches and replaces the other instances

Comment (Borda) [reply lib/matplotlib/axes/_base.py:1502]: I can do it tomorrow...

Comment (Borda) [reply lib/matplotlib/axes/_base.py]: btw, there are also many cases like
```
color : color
	Color for axis pane.
```
shall this be also extended?
```
color : color; see :ref:`colors_def`
	Color for axis pane.
```
or
```
color : color
	Color for axis pane. See :ref:`colors_def`.
```

**Comment by timhoffm** [comment]
Please wait for #27200.

**Review by jklymak — APPROVED** [review]

**Comment by jklymak** [comment]
I don't think this needs to wait for #27200 - it's a simple search and replace if we decide to add a new custom role, whereas this greatly improves things with or without that feature.

**Review by timhoffm — APPROVED** [review]

**Pull request merged**